common/src/main/java/com/microsoft/identity/common/internal/msafederation/FederatedCredential.kt

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.msafederation
+
+/**
+ * Represents credential artifact as result of successful sign in into a federated sign in provider
+ * (Google/Apple). It can contain id token and/or auth code. See implementations for more details.
+ */
+abstract class FederatedCredential(val federatedSignInProviderName: FederatedSignInProviderName)
```

Comment:
Find it strange to have a class without any methods at all. Usually an abstract class is declared to have 1 or more abstract methods that can be overridden but this has no methods at all. What are we gaining here?

Reply:
There's a getter method on this. I could do a follow up PR if the abstraction is evaluated extraneous, but idea was to have base class to have notion of sign-in -params and credentials.